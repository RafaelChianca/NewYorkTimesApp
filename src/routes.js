import * as React from 'react';
import { NavigationContainer, TabActions } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import * as Unicons from '@iconscout/react-native-unicons';
import Home from './pages/Home';
import Science from './pages/Science';

const Tab = createBottomTabNavigator();

function Routes() {
    return (
        <NavigationContainer>
             <Tab.Navigator
                initialRouteName="Home"
                tabBarOptions={{
                    keyboardHidesTabBar: true,
                    style: {
                        backgroundColor: 'white',
                        borderTopColor: 'rgba(255, 255, 255, 0.2)',
                        borderTopWidth: 1,
                    },
                    showLabel: true,
                    activeTintColor: 'black'
                }}
            >
                <Tab.Screen 
                    name="Tech"
                    component={Home}
                    options={{
                        tabBarIcon: ({ focused }) => (
                            <Unicons.UilLightbulbAlt size={28} color={focused ? 'black' : 'lightgrey'}/>
                        ),
                    }}
                />
                <Tab.Screen 
                    name="Science"
                    component={Science}
                    options={{
                        tabBarIcon: ({ focused }) => (
                            <Unicons.UilMicroscope size={28} color={focused ? 'black' : 'lightgrey'}/>
                        ),
                    }}
                />
            </Tab.Navigator>
            
        </NavigationContainer>
    );
}

export default Routes;