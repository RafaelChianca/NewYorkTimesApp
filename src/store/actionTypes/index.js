export * from './technology'
export * from './science'