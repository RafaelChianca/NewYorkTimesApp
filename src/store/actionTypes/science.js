export const scienceTypes = {
  REQUESTED: 'LIST_SCIENCE_REQUESTED',
  SUCCEEDED: 'LIST_SCIENCE_SUCCEEDED',
  FAILED: 'LIST_SCIENCE_FAILED',
}
