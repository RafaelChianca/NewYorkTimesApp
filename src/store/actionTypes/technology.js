export const technologyTypes = {
  REQUESTED: 'LIST_TECHNOLOGY_REQUESTED',
  SUCCEEDED: 'LIST_TECHNOLOGY_SUCCEEDED',
  FAILED: 'LIST_TECHNOLOGY_FAILED',
}
