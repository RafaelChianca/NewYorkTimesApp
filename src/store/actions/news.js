import { scienceTypes, technologyTypes } from "../actionTypes"

export const listScienceRequested = (page, limit) => ({
    type: scienceTypes.REQUESTED,
    payload: { page, limit },
})

export const listScienceSucceeded = (scienceList) => ({
    type: scienceTypes.SUCCEEDED,
    payload: { scienceList },
})

export const listScienceFailed = () =>({
    type: scienceTypes.FAILED,
})

export const listTechnologyRequested = (page, limit) => ({
    type: technologyTypes.REQUESTED,
    payload: { page, limit },
})

export const listTechnologySucceeded = (technologyList) => ({
    type: technologyTypes.SUCCEEDED,
    payload: { technologyList },
})

export const listTechnologyFailed = () =>({
    type: technologyTypes.FAILED,
})
