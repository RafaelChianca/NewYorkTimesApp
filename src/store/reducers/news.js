import { technologyTypes, scienceTypes } from "../actionTypes";

const INITIAL_STATE = {
    error: false,
    loading: false,
    technology: [],
    science: [],
}

function news(state = INITIAL_STATE, action) {

    switch (action.type) {
        case technologyTypes.REQUESTED:
            return { ...state, loading: true }
        case technologyTypes.SUCCEEDED:
            return { ...state, error: false, loading: false, technology: action.payload.technologyList }
        case technologyTypes.FAILED:
            return { ...state, error: true, loading: false }
        case scienceTypes.REQUESTED:
            return { ...state, loading: true }
        case scienceTypes.SUCCEEDED:
            return { ...state, error: false, loading: false, science: action.payload.scienceList }
        case scienceTypes.FAILED:
            return { ...state, error: true, loading: false }
        default:
            return state;
    }
}

export default news;
