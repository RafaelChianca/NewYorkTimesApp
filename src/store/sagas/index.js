import { Alert } from 'react-native';
import { call, all, put, takeLatest } from 'redux-saga/effects'
import api from '../../services/api';
import { scienceTypes, technologyTypes } from '../actionTypes';
import * as NewsActions from '../actions/news';
import { apiKey } from '../../settings/apiSetup';

function * listTechnologyNews ( action ) {
    try {
        const response = yield call(api.get,`technology.json?api-key=${apiKey}`);
        yield put(NewsActions.listTechnologySucceeded(response.data.results));
    } catch (error) {
        Alert.alert("Error loading list!", error.message);
        yield put(NewsActions.listTechnologyFailed());
    }
}

function * listScienceNews ( action ) {
    try {
        const response = yield call(api.get,`science.json?api-key=${apiKey}`);
        yield put(NewsActions.listScienceSucceeded(response.data.results));
    } catch (error) {
        Alert.alert("Error loading list!", error.message);
        yield put(NewsActions.listScienceFailed());
    }
}

function * mySaga () {
    return yield all ([
        takeLatest(technologyTypes.REQUESTED, listTechnologyNews),
        takeLatest(scienceTypes.REQUESTED, listScienceNews),
    ])
}

export default mySaga;