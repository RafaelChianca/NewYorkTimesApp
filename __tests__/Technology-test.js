/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { technologyTypes } from '../src/store/actionTypes';
import news from '../src/store/reducers/news';
import Technology from '../src/pages/Technology';
import { Provider } from 'react-redux';
import { store } from '../src/store';

jest.useFakeTimers();

describe('Technology Reducer', () => {
  it('Should return dafault state', () => {
    const newState = news(undefined, {});
    expect(newState).toEqual({
      error: false,
      loading: false,
      technology: [],
      science: [],
    });
  });

  it('Should return new state if receiving type', () => {
    const posts = [{ title: 'test1' }, { title: 'test2' }, { title: 'test3' }]
    const newState = news(undefined, {
      type: technologyTypes.SUCCEEDED,
      payload: {technologyList: posts}
    });
    expect(newState.technology).toEqual(posts);
  });
});

describe('Renders Technology page correctly', () => {
  it('Should render without crashing', () => {
    renderer.create(
      <Provider store={store}>
        <Technology/>
      </Provider>
    );
  })
});
