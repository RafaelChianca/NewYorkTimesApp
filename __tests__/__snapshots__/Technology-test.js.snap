// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technology page Snapshot test Should be equal to snapshot 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 20,
        },
      ]
    }
  >
    Technology
  </Text>
  <RCTScrollView
    ListEmptyComponent={
      <Styled(Component)
        onPress={[Function]}
      >
        <Styled(Text)>
          Click to reload
        </Styled(Text)>
      </Styled(Component)>
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
                "textTransform": "capitalize",
              },
            ]
          }
        >
          Click to reload
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
